controller/lib/src/etcd_client/interface.rs

```diff
@@ -0,0 +1,131 @@
+use etcd_client::{Client, DeleteResponse, Error, GetOptions, PutResponse};
+use log::{debug, error, info};
+
+pub struct EtcdInterface {
+  client: Client,
+}
```

Comment:
So this is mostly a wrapper over the `etcd-client` crate APIs and e.g. this structure is a `Client` convenience.

Please apply the following changes:

1. Rename your `etcd_client` crate to `etcd`
2. Keep everything under the `mod.rs` file, this is really just a convenience API.
3. Rename `EtcdInterface` to `EtcdClient`:
```rust
pub struct EtcdClient {
    inner: Client,
}
```

Reply:
Thanks, I have updated files regarding to your remarks :+1: